framework_tvb/tvb/interfaces/web/controllers/simulator_controller.py

```diff
@@ -1122,3 +1114,49 @@ def export(self, burst_id):
 
         result_name = "tvb_simulation_" + str(burst_id) + ".zip"
         return serve_file(export_zip, "application/x-download", "attachment", result_name)
+
+    @expose_fragment("overlay")
+    def get_upload_overlay(self):
+        template_specification = self.fill_overlay_attributes(None, "Upload", "Simulation ZIP",
+                                                              "burst/upload_burst_overlay", "dialog-upload")
+        template_specification['first_fragment_url'] = SimulatorWizzardURLs.SET_CONNECTIVITY_URL
+        return self.fill_default_attributes(template_specification)
+
+    @cherrypy.expose
+    @handle_error(redirect=True)
+    @check_user
+    @settings
+    def load_simulator_configuration_from_zip(self, **data):
+        """Upload Simulator from previously exported ZIP file"""
+        self.logger.debug("Uploading ..." + str(data))
+
+        try:
+            upload_param = "uploadedfile"
+            if upload_param in data and data[upload_param]:
+                import_service = ImportService()
+                simulator_folder = import_service.import_simulator_configuration_zip(data[upload_param])
+
+                project = common.get_current_project()
+                simulator_h5_filename = DirLoader(simulator_folder, None).find_file_for_has_traits_type(Simulator)
+                with SimulatorH5(os.path.join(simulator_folder, simulator_h5_filename)) as sim_h5:
+                    simulator_gid = sim_h5.gid.load()
+                simulator = SimulatorSerializer.deserialize_simulator(simulator_gid, simulator_folder)
+
+                bc_h5_filename = DirLoader(simulator_folder, None).find_file_for_has_traits_type(BurstConfiguration)
+                burst_config = BurstConfiguration(project.id)
+                with BurstConfigurationH5(os.path.join(simulator_folder, bc_h5_filename)) as bc_h5:
+                    bc_h5.load_into(burst_config)
```

Comment:
Except for the session related code, and parsing of input parameters, the rest of the code would better sit in a service (e.g. everything dealing with DirLoader, or h5.load_into...)